### PR TITLE
Test Stubs for 6.3 feature Canned Role

### PR DESCRIPTION
**Pull request by jyejare**
Test Stubs for Satellite 6.3 feature Canned Role Org Admin
For all forms,
UI,
CLI,
API

They are 40 stubs for all forms.

### CODE REVIEW

**Comment by jyejare** [comment]
More stubs will be added in next PR.

File: tests/foreman/api/test_role.py [hunk 1]
```diff
@@ -80,3 +80,186 @@ def test_positive_update(self):
                 role = entities.Role().create()
                 role.name = name
                 self.assertEqual(role.update(['name']).name, name)
+
+
+class CannedRoleTestCases(APITestCase):
+    """Implements Canned Roles tests from UI"""
+
+    @tier1
```
Comment (sghai) [inline tests/foreman/api/test_role.py]: @stubbed ? This is applicable for all tests.

File: tests/foreman/api/test_role.py [hunk 2]
```diff
@@ -80,3 +80,186 @@ def test_positive_update(self):
...
+        @caseautomation: notautomated
+        """
+
+    @tier1
+    def test_positive_create_filter_without_override(self):
+        """Create filter in role w/o overriding it
+
+        @id: 1aadb7ea-ff76-4171-850f-188ba6f87021
+
+        @steps:
```
Comment (sghai) [inline tests/foreman/api/test_role.py]: one line space is needed after @steps:

File: tests/foreman/api/test_role.py [hunk 3]
```diff
@@ -80,3 +80,186 @@ def test_positive_update(self):
...
+        1. Create a role with taxonomies assigned
+        2. Create filter in role without overriding it
+
+        @assert:
```
Comment (sghai) [inline tests/foreman/api/test_role.py]: empty line needed after @assert

File: tests/foreman/api/test_role.py [hunk 4]
```diff
@@ -80,3 +80,186 @@ def test_positive_update(self):
...
+        1. Create role with overridden filter having different taxonomies than
+        its role.
+        2. Unset the override flag in above role filter
+
+        @assert: The taxonomies of filters resets/synced to role taxonomies
+
+        @caseautomation: notautomated
+        """
+
+    @tier1
```
Comment (sghai) [inline tests/foreman/api/test_role.py]: should be a Tier2 test

File: tests/foreman/api/test_role.py [hunk 5]
```diff
@@ -80,3 +80,186 @@ def test_positive_update(self):
...
+        3. Create user with taxonomies same as role taxonomies
+        4. Assign step 1 role to user
+
+        @assert: User should be able to access the resource(s) of the assigned
+        role
+
+        @caseautomation: notautomated
+        """
+
+    @tier1
```
Comment (sghai) [inline tests/foreman/api/test_role.py]: tier2 ?

File: tests/foreman/cli/test_role.py [hunk 6]
```diff
@@ -123,3 +123,189 @@ def test_positive_update_name(self):
...
+        1. Create role with overridden filter having different taxonomies than
+        its role.
+        2. Unset the override flag in above role filter
+
+        @assert: The taxonomies of filters resets/synced to role taxonomies
+
+        @caseautomation: notautomated
+        """
+
+    @tier1
```
Comment (sghai) [inline tests/foreman/cli/test_role.py]: tier2 ?

File: tests/foreman/cli/test_role.py [hunk 7]
```diff
@@ -123,3 +123,189 @@ def test_positive_update_name(self):
...
+        2. Create resource(s) filter(s) without overriding them
+        3. Create user with taxonomies same as role taxonomies
+        4. Assign step 1 role to user
+
+        @assert: User should be able to access the resource(s) of the assigned
+        role
+
+        @caseautomation: notautomated
+
+        @CaseLevel: System
```
Comment (sghai) [inline tests/foreman/cli/test_role.py]: @CaseLevel: system is valid for tier3  not tier1
and for tier2, @caselevel should be Integration.

File: tests/foreman/ui/test_role.py [hunk 8]
```diff
@@ -129,3 +129,315 @@ def test_positive_update_org(self):
...
+        2. Create resource(s) filter(s) without overriding them
+        3. Create user with taxonomies not matching role taxonomies
+        4. Assign step 1 role to user
+
+        @assert: User should not be able to access the resource(s) that are not
+        associated to assigned role
+
+        @caseautomation: notautomated
+
+        @CaseLevel: System
```
Comment (sghai) [inline tests/foreman/ui/test_role.py]: for tier2, @CaseLevel: should be Integration

**Comment by sghai** [comment]
Common comments for all tests:

- empty line needed between  @steps and step1
- empty line needed between @assert and step1
- @stubbed needs to be set on all tests
- for tiet2, set CaseLevel: Integration

run `make docs-clean docs` once to see the generated layout of tests doctrings

**Comment by jyejare** [comment]
@sghai I finished working on your comments and the PR is updated. Please have a look.

**Comment by sghai** [comment]
ACK

**Comment by renzon** [comment]
ACK

**Pull request merged**